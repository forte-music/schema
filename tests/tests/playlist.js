import gql from 'graphql-tag';
import { print } from 'graphql/language/printer';

import PlaylistFields from './fragments/PlaylistFields.graphql';
import PlaylistItemFields from './fragments/PlaylistItemFields.graphql';
import SongFields from './fragments/SongFields.graphql';

import client from '../client';

it('should resolve a query for data stemming from a playlist', async () => {
  const variables = { playlistId: 'playlist:1' };

  const query = gql`
    query($playlistId: ID!) {
      playlist(id: $playlistId) {
        ...PlaylistFields

        items(input: { limit: 10 }) {
          edges {
            node {
              ...PlaylistItemFields

              song {
                ...SongFields
              }
            }
          }
        }
      }
    }

    ${PlaylistFields}
    ${PlaylistItemFields}
    ${SongFields}
  `;

  const resp = await client.request(print(query), variables);
  expect(resp).toMatchSnapshot();
});
