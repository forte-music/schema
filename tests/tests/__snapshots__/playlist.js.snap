// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should resolve a query for data stemming from a playlist 1`] = `
Object {
  "playlist": Object {
    "description": "",
    "duration": 481,
    "id": "playlist:1",
    "items": Object {
      "edges": Array [
        Object {
          "node": Object {
            "id": "playlist:1:0",
            "song": Object {
              "diskNumber": 1,
              "id": "song:2:1",
              "name": "I'm That (Remix)",
              "streamUrl": "/music/01 - I'm That... (Remix) [feat. Beenie Man & Azealia Banks].m4a",
              "trackNumber": 1,
            },
          },
        },
        Object {
          "node": Object {
            "id": "playlist:1:1",
            "song": Object {
              "diskNumber": 1,
              "id": "song:3:1",
              "name": "Bugatti",
              "streamUrl": "/music/01 - Bugatti (feat. Future & Rick Ross).flac",
              "trackNumber": 1,
            },
          },
        },
      ],
    },
    "name": "Quick Jam Session",
  },
}
`;
